### PR TITLE
[dy] Add db connection url environment variable

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Add environment variables so that users can change what DB is used to store run data.

URL format: `postgresql+psycopg2://user:password@host:port/dbname`

The postgresql connector `psycopg2` will error out if the connection is used by multiple processes. I updated the backend code to create a new session for each process that is created. Also, timestamps are handled differently by postgres vs sqlite, so I changed how dates are compared.

# Tests

tested with rds postgres db

cc: @wangxiaoyou1993

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Delete the empty file? `mage_ai/orchestration/db/db_config.py`

**Review by wangxiaoyou1993 — APPROVED** [review]

**Comment by dy46** [comment]
@wangxiaoyou1993 there's an issue with psycopg and multiprocessing that i'm trying to fix, so i'll be adding a bit to this pr

**Comment by wangxiaoyou1993** [comment]
> @wangxiaoyou1993 there's an issue with psycopg and multiprocessing that i'm trying to fix, so i'll be adding a bit to this pr

Ok. You can ping me when you need another review.

**Comment by dy46** [comment]
@wangxiaoyou1993 PTAL. I made some small frontend bug fixes, but most of the significant changes are in the backend files.

File: mage_ai/shared/dates.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import datetime
+import pytz
+
+def compare(date1: datetime, date2: datetime) -> int:
+    date1_str = date1.strftime(format='%Y%m%dT%H%M%S')
```
Comment (wangxiaoyou1993) [inline mage_ai/shared/dates.py:5]: does this ignore timezone?

File: mage_ai/orchestration/triggers/loop_time_trigger.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+from mage_ai.orchestration.db import db_connection
```
Comment (wangxiaoyou1993) [inline mage_ai/orchestration/triggers/loop_time_trigger.py:1]: is it needed?

File: scripts/deploy/terraform/aws/variables.tf [hunk 3]
```diff
@@ -8,6 +8,10 @@ variable "AWS_SECRET_ACCESS_KEY" {
   default = "AWS_SECRET_ACCESS_KEY"
 }
 
+variable "DATABASE_CONNECTION_URL" {
```
Comment (wangxiaoyou1993) [inline scripts/deploy/terraform/aws/variables.tf:11]: Provide a default value here. This should be an optional variable.

File: mage_ai/server/scheduler_manager.py [hunk 4]
```diff
@@ -6,10 +6,14 @@
 
 SCHEDULER_AUTO_RESTART_INTERVAL = 10
 
+from mage_ai.orchestration.db.process import create_process
```
Comment (wangxiaoyou1993) [inline mage_ai/server/scheduler_manager.py:9]: why do you put the import statement under the constant?

File: mage_ai/orchestration/pipeline_scheduler.py [hunk 5]
```diff
@@ -212,8 +216,16 @@ def __schedule_pipeline(self) -> None:
             f'Start a process for PipelineRun {self.pipeline_run.id}',
             **self.__build_tags(),
         )
-
-        proc = multiprocessing.Process(target=run_pipeline, args=(
+        variables = merge_dict(
```
Comment (wangxiaoyou1993) [inline mage_ai/orchestration/pipeline_scheduler.py:219]: These are unused. They are replaced by `self.__get_variables()` below.

File: mage_ai/orchestration/pipeline_scheduler.py [hunk 6]
```diff
@@ -8,7 +8,9 @@
 from mage_ai.data_preparation.models.pipelines.integration_pipeline import IntegrationPipeline
 from mage_ai.data_preparation.repo_manager import get_repo_path
 from mage_ai.data_preparation.variable_manager import get_global_variables
+from mage_ai.orchestration.db import db_connection
```
Comment (wangxiaoyou1993) [inline mage_ai/orchestration/pipeline_scheduler.py:11]: unused import

File: mage_ai/shared/dates.py (hunk 1 above)
Comment (wangxiaoyou1993) [reply mage_ai/shared/dates.py:5]: Could you also handle None error?

File: mage_ai/server/scheduler_manager.py (hunk 4 above)
Comment (wangxiaoyou1993) [reply mage_ai/server/scheduler_manager.py:9]: also remove the unused multiprocessing import.

File: mage_ai/orchestration/pipeline_scheduler.py (hunk 6 above)
Comment (wangxiaoyou1993) [reply mage_ai/orchestration/pipeline_scheduler.py:11]: also remove the unused `multiprocessing` import.

File: mage_ai/orchestration/db/models.py [hunk 7]
```diff
@@ -179,7 +195,7 @@ def should_schedule(self) -> bool:
             current_execution_date = self.current_execution_date()
             if current_execution_date is None:
                 return False
-            if not find(lambda x: x.execution_date == current_execution_date, self.pipeline_runs):
+            if not find(lambda x: compare(x.execution_date, current_execution_date) == 0, self.pipeline_runs):
```
Comment (wangxiaoyou1993) [inline mage_ai/orchestration/db/models.py:198]: nit: break the line if it's too long

File: mage_ai/shared/dates.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import datetime
+import pytz
+
+def compare(date1: datetime, date2: datetime) -> int:
+    if date1 is None or date2 is None:
+        return None
+
+    date1_utc = date1.replace(tzinfo=pytz.UTC)
+    date2_utc = date2.replace(tzinfo=pytz.UTC)
+    date1_str = date1_utc.strftime(format='%Y%m%dT%H%M%S')
```
Comment (wangxiaoyou1993) [inline mage_ai/shared/dates.py:10]: Are we able to compare the datetime objects directly after converting them to UTC?

File: scripts/deploy/terraform/aws/variables.tf [hunk 9]
```diff
@@ -8,6 +8,10 @@ variable "AWS_SECRET_ACCESS_KEY" {
...
+  type    = string
```
Comment (wangxiaoyou1993) [inline scripts/deploy/terraform/aws/variables.tf:12]: Also provide a default value here.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**